### DISCUSSION TITLE
自定义tabbar  必须是class组件吗？ 不能使用function组件吗?

### DISCUSSION THREAD

**Discussion by H-Sven** [NervJS/taro#13719] [Q & A]
发现如果使用function组件的话
const pageObj = getCurrentInstance().page
    const tabBar = getTabBar(pageObj)
    tabBar?.onClickTab('pages/mine/index')

无法获取到onClickTab方法， 请求在function组件中onClickTab方法应该如何暴露出来

**Comment by 18338625417** [comment]
redux或者其它的仓库，去修改tabbar

**Comment by liuyajie110** [comment]
监听路由变化也行 就是闪烁的问题一直解决不了

**Reply by 18338625417** [reply]
微信小程序自定义tabbar闪烁的问题 貌似目前还无法解决 原生也是同样的问题切换会闪烁
